File: ReportesApi/Models/Report.cs
namespace ReportesApi.Models;

public class Report
{
    public int Id { get; set; }
    public string Folio { get; set; } = Guid.NewGuid().ToString();
    public string Title { get; set; } = "";
    public string Description { get; set; } = "";
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public string ImageUrl { get; set; } = "";
    public ReportStatus Status { get; set; } = ReportStatus.Pending;

    public int UserId { get; set; }
    public User User { get; set; } = null!;

    public int? UserReviewerId { get; set; }
    public DateTime? ReviewDate { get; set; }
}

File: ReportesApp/lib/core/models/report.dart
class Report {
  final int id;
  final String title;
  final String folio;
  final String description;
  final String imageUrl;
  final DateTime createdAt;
  final int status;
  final DateTime? reviewDate;
  final String? reviewerName;

  Report({
    required this.id,
    required this.title,
    required this.folio,
    required this.description,
    required this.imageUrl,
    required this.createdAt,
    required this.status,
    this.reviewDate,
    this.reviewerName,
  });

  factory Report.fromJson(Map<String, dynamic> json) {
    return Report(
      id: json['id'],
      title: json['title'],
      folio: json['folio'],
      description: json['description'],
      imageUrl: json['imageUrl'],
      createdAt: DateTime.parse(json['createdAt']),
      status: json['status'],
      reviewDate: json['reviewDate'] != null ? DateTime.parse(json['reviewDate']) : null,
      reviewerName: json['reviewerName'],
    );
  }
}